File: src/lucy.c
/*
 * lucy.c
 *
 * Copyright (C) 2014 - Wiky L
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
#define  TRACE_TAG   TRACE_ADB

#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <stdarg.h>
#include <errno.h>
#include <stddef.h>
#include <string.h>
#include <time.h>
#include <sys/time.h>
#include <stdint.h>

#include "libadb/sysdeps.h"
#include "libadb/adb.h"
#include "libadb/adb_auth.h"
#include "libadb/usb_vendors.h"

#include "ui_mainwindow.h"
#include "lcutil.h"


int main(int argc, char **argv)
{
    adb_sysdeps_init();
    adb_trace_init();
    if (g_strcmp0(argv[0], "adb") == 0) {
        /*
         * if argv[0] is "adb", lucy runs as adb
         */
        return adb_commandline(argc - 1, argv + 1);
    }
    lc_init(argc, argv);
    gtk_init(&argc, &argv);

    UIMainWindow *window = ui_main_window_new();
    ui_main_window_show(window);

    return 0;
}

File: bin/lucy.py
#!/usr/bin/env python3
# encoding=utf8

# The lucy executable script.
#
# Copyright (C) 2015 Wiky L
#
# This copyrighted material is made available to anyone wishing to use,
# modify, copy, or redistribute it subject to the terms and conditions of
# the GNU General Public License v.3, or (at your option) any later version.
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY expressed or implied, including the implied warranties of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
# Public License for more details.  You should have received a copy of the
# GNU General Public License along with this program; if not, write to the
# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
# 02110-1301, USA.
#

import os
import sys

if __name__ != '__main__':
    sys.exit(1)


try:
    from pdb import *
    from pydroid.main import pydroid_main
except ImportError:
    sys.path.append('.')
    sys.path.append('./pdb/')
    print(sys.path)
    from pdb import *
    print(dir())
    from pydroid.main import pydroid_main

pydroid_main()
